foundryup/install

```diff
@@ -37,10 +38,42 @@ OPTIONS:
     -h, --help      Print help information
     -v, --version   Install a specific version
     -b, --branch    Install a specific branch
-    -r, --repo      Install a forks main branch"
+    -r, --repo      Install a forks main branch
+    -p, --path      Install a local repository"
     exit 0
 fi
 
+# Installs foundry from a local repository if --path parameter is provided
+if [[ -n "$FOUNDRYUP_LOCAL_REPO" ]]; then
+
+  # Ignore branches/versions as we do not want to modify local git state
+  if [ -n "$FOUNDRYUP_REPO" ] || [ -n "$FOUNDRY_BRANCH" ] || [ -n "$FOUNDRY_VERSION" ]; then
+    echo "foundryup: !WARN! --branch, --version, and --repo arguments are ignored during local install"
+  fi
+
+  if ! command -v cargo &> /dev/null ; then
+    # Error if cargo is not already installed.
+    echo "foundryup: cargo is not installed. Please install it first."
+    exit 1
+  fi
+
+  # Enter local repo and build
+  FOUNDRY_REPO="$PWD/$FOUNDRYUP_LOCAL_REPO"
```

Comment:
This will fail if you use an absolute path. I think just using `$FOUNDRYUP_LOCAL_REPO` should handle both cases?

Reply:
agreed - would prefer we use absolute paths, shell expansion should cover converting it otherwise

Reply:
agreed! This was left over from some relative-path symlinking debugging. Pushed changes that following @onbjerg suggestion and confirmed it supports both absolute and relative paths. Also added some testing notes in the PR.